tests/integration/sharding_tests/helpers.py

```diff
@@ -15,6 +16,25 @@
 SHARD_REL_NAME = "sharding"
 CLUSTER_COMPONENTS = [SHARD_ONE_APP_NAME, SHARD_TWO_APP_NAME, CONFIG_SERVER_APP_NAME]
 TIMEOUT = 15 * 60
+MONGOS_PORT = 27018
+MONGOD_PORT = 27017
+
+
+def count_users(mongos_client: MongoClient) -> int:
+    """Returns the number of users using the cluster."""
+    admin_db = mongos_client["admin"]
+    users_collection = admin_db.system.users
+    return users_collection.count_documents({})
+
+
+async def get_username_password(ops_test: OpsTest, app_name: str, relation_name: str) -> Tuple:
```

Comment:
nit (this works for getting integrated usernames but not the admin/operator username/password

```suggestion
async def get_related_username_password(ops_test: OpsTest, app_name: str, relation_name: str) -> Tuple:
       """Retrieves the username and password for an integrated application using app_name and relation_name"""
```